analyzer/consensus/consensus.go

```diff
@@ -299,25 +301,29 @@ func (m *processor) ProcessBlock(ctx context.Context, uheight uint64) error {
 		return fmt.Errorf("height %d is too large", uheight)
 	}
 	height := int64(uheight)
+	batch := &storage.QueryBatch{}
 
-	// Fetch all data.
-	fetchTimer := m.metrics.BlockFetchLatencies()
-	data, err := fetchAllData(ctx, m.source, m.network, height, m.mode == analyzer.FastSyncMode)
-	fetchTimer.ObserveDuration()
-	if err != nil {
-		if strings.Contains(err.Error(), fmt.Sprintf("%d must be less than or equal to the current blockchain height", height)) {
-			return analyzer.ErrOutOfRange
+	isBlockAbsent := m.chain == common.ChainNameMainnet && height == 8048955
```

Comment:
I feel like this could go as some kind of addition to the History config. this is the height just below when mainnet damask started. maybe we could put something in the mainnet cobalt record to say height 8048955 was absent

Reply:
That was my first instinct too. But Jernej was saying how this was definitely a bug/mistake when doing the Damask dump-restore. In the future, a) dump-restores are unlikely, b) if they do happen, we're unlikely to make the same mistake twice. So it's IMO an overgeneralization to bloat the history config with this, and make the consuming code (= here) more complex and general and indirect. Let's expand History once we have more than one such block in the future.

I also thought about at least referencing the Damask genesis height from History config here. So instead of comparing to 8048955, I could compare to (genesis_height - 1). But then I decided that's no more informative, and just harder to grep for.

Reply:
opposite opinion from the same principles:
- it'll take at least a little code change no matter what
- it happened once and we anticipate not again, so this only exists as a historical artifact
- better to shove the details into the "history" data rather than bloat the code with special (name, height) coordinates

ok on not optimizing generality or auto computing the height though

Reply:
this would prevent us from having to add this special purpose chain name arg throughout

Reply:
extra thing: I'm against doing special behavior for named default chains that a custom history config can't do